### PR TITLE
[Tune] Added default values for utility kwargs

### PR DESCRIPTION
**Pull request by LucaCappelletti94**
## Why are these changes needed?
Previously, it was not possible to instantiate a `BayesOptSearch` object without leaving the default parameters for the utility function, as it is possible with the [BayesianOptimization package](https://github.com/fmfn/BayesianOptimization/blob/857f0fe05471db35510a6b03463033ad05b7f4f9/bayes_opt/bayesian_optimization.py#L159). I have now added the default parameters when no others are provided.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by LucaCappelletti94** [comment]
I expect there to be some collisions between this pull request and [this other one](https://github.com/ray-project/ray/pull/8486), but they should be quite easy to solve.

**Comment by richardliaw** [comment]
FYI, maybe this code might be useful? https://github.com/ray-project/ray/pull/8216/files

I haven't had time to clean up that PR (it isn't passing tests), but looks very similar to what you're trying to do.

Happy to merge yours instead!

File: python/ray/tune/suggest/bayesopt.py [hunk 1]
```diff
@@ -76,6 +74,15 @@ def __init__(self,
             max_concurrent=max_concurrent,
             use_early_stopped_trials=use_early_stopped_trials)
 
+        if utility_kwargs is None:
+            # The defaults arguments are the same
+            # as in the package BayesianOptimization
+            utility_kwargs = dict(
+                kind="ucb",
+                kappa=2.576,
+                xi=0.0,
```
Comment (richardliaw) [inline python/ray/tune/suggest/bayesopt.py:83]: Can you add a note in the docstring that the values are defaulted to this?

Comment (LucaCappelletti94) [reply python/ray/tune/suggest/bayesopt.py:83]: Sure!

**Review by richardliaw — APPROVED** [review]

**Pull request merged**